### PR TITLE
Small fixes

### PR DESCRIPTION
**Pull request by drakmor**
Fixed errors after static code analysys

### CODE REVIEW

File: src/modules/bluefish/producer/bluefish_producer.cpp [hunk 1]
```diff
@@ -348,7 +348,7 @@ struct bluefish_producer : boost::noncopyable
             blue_->render_buffer_capture(BlueBuffer_Image_HANC(schedule_capture_frame_id_));
             dma_ready_captured_frame_id_ = capturing_frame_id_;
             capturing_frame_id_          = schedule_capture_frame_id_;
-            schedule_capture_frame_id_   = (++schedule_capture_frame_id_ % 4);
+            schedule_capture_frame_id_   = (schedule_capture_frame_id_ + 1) % 4;
```
Comment (Julusian) [inline src/modules/bluefish/producer/bluefish_producer.cpp]: schedule_capture_frame_id_ is no longer being incremented. Was this intentional?

File: src/modules/decklink/decklink.cpp [hunk 2]
```diff
@@ -60,14 +60,21 @@ std::vector<std::wstring> device_list()
...
             String m_name;
+#endif        
             bool   success = SUCCEEDED(decklink->GetModelName(&m_name));
             decklink->Release();
             std::wstring model_name = L"Unknown";
 
-            if (success)
-                model_name = u16(m_name);
-
+            if (success && m_name != nullptr)
```
Comment (Julusian) [inline src/modules/decklink/decklink.cpp]: Can this not use the get_model_name function defined in util.h?

File: src/modules/decklink/util/util.h [hunk 3]
```diff
@@ -252,9 +252,17 @@ static com_ptr<IDeckLink> get_device(size_t device_index)
 template <typename T>
 static std::wstring get_model_name(const T& device)
 {
+
```
Comment (Julusian) [inline src/modules/decklink/util/util.h]: Why was this change needed?
It was much cleaner before without any ifdefs, and worked as far as Im aware.

**Comment by Julusian** [comment]
I'm curious, what did you use to do this static analysis?

File: src/modules/bluefish/producer/bluefish_producer.cpp (hunk 1 above)
Comment (drakmor) [reply src/modules/bluefish/producer/bluefish_producer.cpp]: schedule_capture_frame_id_  was previously increased, and then overwritten by a new value. The modification of a variable is unsequenced relative to another operation on the same variable. This may lead to undefined behavior.

File: src/modules/decklink/decklink.cpp (hunk 2 above)
Comment (drakmor) [reply src/modules/decklink/decklink.cpp]: Yes we can

File: src/modules/decklink/util/util.h (hunk 3 above)
Comment (drakmor) [reply src/modules/decklink/util/util.h]: Because different string types in Windows/Linux/Apple
1. https://forum.blackmagicdesign.com/viewtopic.php?p=423675#p423675
2. https://www.viva64.com/en/w/v745/print
3. Like this - https://github.com/Streampunk/macadam/blob/master/src/macadam.cc

**Comment by drakmor** [comment]
> I'm curious, what did you use to do this static analysis?

PVS Studio

File: src/modules/decklink/util/util.h (hunk 3 above)
Comment (Julusian) [reply src/modules/decklink/util/util.h]: Yes, but we don't need to do it via ifdefs in this way.
`String` is a custom typedef in https://github.com/CasparCG/server/blob/master/src/modules/decklink/decklink_api.h that on windows is `BSTR` and on linux is `const char*`

I am assuming there was some static analysis problem here due to BSTR being more than just a byte array and it being treated as such when passed into `u16(`, so how about adding a `std::wstring to_string(String);` for each platform instead in decklink_api.h which this code can use instead of u16?

Comment (drakmor) [reply src/modules/decklink/util/util.h]: It's best idea.

**Comment by drakmor** [comment]
Repo rebased
